### PR TITLE
Support `lts-20.4`.

### PR DESCRIPTION
**Pull request by vikt0r0**
## Purpose

Upgrade to support stackage snapshot to `lts-20.4`.

## Changes

- Bump resolver in `stack.yaml` to `lts-20.4`.
- Update dependency of `proto3-wire` to 1.4.0.
- Update dependency on `http2-client` to one that is updated to support `lts-20.4`.

## Dependencies

https://github.com/Concordium/http2-client/pull/3

## Checklist

- [X] My code follows the style of this project.
- [x] The code compiles without warnings.
- [X] I have performed a self-review of the changes.
- [X] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: bench-tool/bench-tool.cabal [hunk 1]
```diff
@@ -1,10 +1,10 @@
 cabal-version: 1.12
```
Comment (vikt0r0) [inline bench-tool/bench-tool.cabal:1]: Should this file be reverted?

File: stack-20.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+resolver: lts-20.4
+allow-newer: true
```
Comment (abizjak) [inline stack-20.yaml]: Why this?

Comment (vikt0r0) [reply stack-20.yaml]: I copied it from `stack-19.yaml`.

I bumped the resolver to `20.4`, upgraded to `proto3-wire-1.4.0` and the commit hash of `http2-client`.

File: bench-tool/bench-tool.cabal [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 
--- This file has been generated from package.yaml by hpack version 0.34.6.
+-- This file has been generated from package.yaml by hpack version 0.35.0.
```
Comment (MilkywayPirate) [inline bench-tool/bench-tool.cabal:3]: I presume this change is because of a stack bump. In particular 2.7.5 -> 2.9.1. 
Is this required? If not, is it on purpose?

File: http2-grpc-proto3-wire/http2-grpc-proto3-wire.cabal [hunk 4]
```diff
@@ -1,10 +1,10 @@
 cabal-version: 1.12
 
--- This file has been generated from package.yaml by hpack version 0.33.0.
+-- This file has been generated from package.yaml by hpack version 0.35.0.
```
Comment (MilkywayPirate) [inline http2-grpc-proto3-wire/http2-grpc-proto3-wire.cabal:3]: Is this change related to the ghc bump?

File: http2-grpc-types/http2-grpc-types.cabal [hunk 5]
```diff
@@ -1,10 +1,10 @@
 cabal-version: 1.12
 
--- This file has been generated from package.yaml by hpack version 0.31.2.
+-- This file has been generated from package.yaml by hpack version 0.35.0.
```
Comment (MilkywayPirate) [inline http2-grpc-types/http2-grpc-types.cabal:3]: Same comment as above.

**Review by MilkywayPirate — COMMENTED** [review]
Looks good
I'm not quite sure I understand why the cabal files are changed? Is that due to bumping the ghc version?

File: bench-tool/bench-tool.cabal (hunk 3 above)
Comment (vikt0r0) [reply bench-tool/bench-tool.cabal:3]: It's not on purpose, it was what was generated using the stack version speciefied in lts-20.4, i.e. Stack @ 2.9.1. I think it makes sense to regenerate the file since the dependencies were bumped. Since it's a generated  file, it seems dodgy to edit it by hand - is the line in question problematic however?

File: http2-grpc-proto3-wire/http2-grpc-proto3-wire.cabal (hunk 4 above)
Comment (vikt0r0) [reply http2-grpc-proto3-wire/http2-grpc-proto3-wire.cabal:3]: Stack @ 2.9.1 bumps hpack to 0.35, and to my understanding this is the version supported by lts-20.4, see https://www.stackage.org/lts-20.4. Is this change problematic?
